### PR TITLE
Polars: shuffle OnceFrame output

### LINKED ISSUE #1826: Polars: shuffle OnceFrame output
The ordering of records in the dataset is also sensitive information, and it can be indirectly observed via the ordering of rows in groupby queries.

Lazyframes themselves do not support shuffling, so this must be done once collected in onceframes.

### PR DESCRIPTION
**Pull request by Shoeboxam**
Close #1826

### CODE REVIEW

**Comment by Shoeboxam** [comment]
* **#1827** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1827?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by mccalluc — COMMENTED** [review]
Suggest keeping the code-formatting on `honest-but-curious`... unless that's an intentional change?

Another question about whether `lazy` is even the right thing to offer, but that shouldn't block this.

File: rust/src/data/ffi/polars.rs [hunk 1]
```diff
@@ -39,8 +39,9 @@ pub extern "C" fn opendp_data__onceframe_collect(
 /// Each collection consumes the entire allocated privacy budget.
 /// To remain DP at the advertised privacy level, only collect the LazyFrame once.
 ///
-/// # Features
-/// * `honest-but-curious` - LazyFrames can be collected an unlimited number of times.
+/// Requires "honest-but-curious" because the privacy guarantees only apply if:
+/// 1. The LazyFrame (compute plan) is only ever executed once.
+/// 2. The analyst does not observe ordering of rows in the output. To ensure this, shuffle the output.
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply rust/src/data/ffi/polars.rs:44]: This is how I anticipate integrating with SmartNoise-SDK SQL-- they would extract the lazyframe plan and then traverse it to run the computations on other backends.

**Comment by Shoeboxam** [comment]
@mccalluc I shouldn't have modified! Only true diff is to accept your suggestion.

**Review by mccalluc — APPROVED** [review]
Thanks!

**Pull request merged**